src/pymmcore_widgets/_hcs_widget/_main_hcs_widget.py

```diff
@@ -0,0 +1,596 @@
+import warnings
+from pathlib import Path
+from typing import List, Optional, Tuple, Union
+
+import numpy as np
+import yaml  # type: ignore
+from pymmcore_plus import CMMCorePlus
+from pymmcore_plus.mda import PMDAEngine
+from qtpy.QtCore import Qt
+from qtpy.QtGui import QBrush
+from qtpy.QtWidgets import QFileDialog, QTableWidgetItem, QWidget
+from superqt.utils import signals_blocked
+from useq import MDASequence
+
+from ._graphics_items import FOVPoints, Well
+from ._main_hcs_gui import HCSGui
+from ._update_yaml_widget import UpdateYaml
+from ._well_plate_database import WellPlate
+
+PLATE_DATABASE = Path(__file__).parent / "_well_plate.yaml"
+AlignCenter = Qt.AlignmentFlag.AlignCenter
+
+
+class HCSWidget(HCSGui):
+    """HCS widget.
+
+    Parameters
+    ----------
+    parent : Optional[QWidget]
+        Optional parent widget, by default None
+    include_run_button: bool
+        By default, False. If true, a "run" button is added to the widget.
+        The acquisition defined by the `useq.MDASequence` built through the
+        widget is executed when clicked.
+    mmcore: Optional[CMMCorePlus]
+        Optional `CMMCorePlus` micromanager core.
+        By default, None. If not specified, the widget will use the active
+        (or create a new) `CMMCorePlus.instance()`.
+
+    The `HCSWidget` provides a GUI to construct a `useq.MDASequence` object.
+    It can be used to automate the acquisition of multi-well plate
+    or custom defined areas.
+    If the `include_run_button` parameter is set to `True`, a "run" button is added
+    to the GUI and, when clicked, the generated `useq.MDASequence` is passed to the
+    `CMMCorePlus.run_mda` method and the acquisition is executed.
+    """
+
+    def __init__(
+        self,
+        parent: Optional[QWidget] = None,
+        *,
+        include_run_button: bool = False,
+        mmcore: Optional[CMMCorePlus] = None,
+    ) -> None:
+        super().__init__(parent)
+
+        self._set_enabled(False)
+
+        self._include_run_button = include_run_button
+
+        self.cancel_Button.hide()
+        self.pause_Button.hide()
+        if not self._include_run_button:
+            self.run_Button.hide()
+
+        self.wp: WellPlate = None  # type: ignore
```

Comment:
as a general comment, you're using `# type: ignore` far too often.  There are around 40 type ignores in this PR, most of which are telling you something useful.  It would be better to simply turn off mypy instead of adding type ignore everywhere (but better still to type it correctly)